qiita_db/search.py

```diff
@@ -176,6 +177,11 @@ def __call__(self, searchstr, user):
             Search string to use
         user : User object
             User making the search. Needed for permissions checks.
+        remove_selected : bool, optional
+            If search in on an analysis that already has selected samples,
+            remove the selected samples from the results. Default False.
```

Comment:
If you have an analysis with samples previously selected, this will remove those samples from the results returned. it keeps you from doing two searches and selecting the same sample twice because it showed up in both searches. This was originally filtered at the python level, but filtering it at the SQL level is faster.